### ISSUE TITLE
Run nixos tests for each PR

### ISSUE THREAD

**Issue by DavHau** [NixOS/nix#9677]
**Is your feature request related to a problem? Please describe.**
When a PR breaks a nixos test, there is no feedback about it and it doesn't block the merge.

**Describe the solution you'd like**
Provide fast feedback about any breaking test to the author of a PR as well as the reviewers.

**Describe alternatives you've considered**
No quick feedback about breaking nixos tests
  -> reduced confidence in changes
  -> increased overhead for reviewers
  -> harder to track down cause once breakage gets discovered.

**Additional context**
What's a quick and easy way to get Ci checks for the nixos tests?

Ideas:
- use hydra and connect to github somehow
- rent hosted third party github runners with KVM (not sure where)
- self host github runners
- use hercules (cc @roberth )
- use garnix
- use https://github.com/Mic92/buildbot-nix (cc @Mic92 )

**Priorities**

Add :+1: to [issues you find important](https://github.com/NixOS/nix/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).

**Comment by Mic92** [comment]
Github's own runner now also supports kvm btw.

**Comment by DavHau** [comment]
> Github's own runner now also supports kvm btw.

Nice, thanks for the hint. I just [tried out running all the tests in a runner](https://github.com/DavHau/nix/actions/runs/7363563910/job/20043140249).
It seems to be unacceptably slow. I mean it already was before enabling the tests. Compiling nix alone takes around 15 minutes on the linux runner. What can we do to improve this situation?

**Comment by Mic92** [comment]
I don't think we can improve nix compilation time magically. Regarding nixos tests, I was expecting no significant slow down since download dependencies should be finished within the 15 minutes window. Can you check if /dev/kvm is present in the build?

**Comment by zowoq** [comment]
Also see the discussion in https://github.com/NixOS/nix/issues/7674.

**Comment by roberth** [comment]
An extra goal - perhaps separate issue? - is to include tests that Nixpkgs thinks are relevant.
We can configure the `nix` flake to run those, with a little bit of collaboration between the repos.
